framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -2218,4 +2220,22 @@ public int getMinLenFromString(String sequenceExpression, Tree tree, TreePath cu
 
         return getMinLenValue(lengthAnno);
     }
+
+    @Override
+    public AnnotatedTypeMirror getAnnotatedType(Tree tree) {
+        AnnotatedTypeMirror atm = super.getAnnotatedType(tree);
+
+        if (atm.isAnnotatedInHierarchy(UNKNOWNVAL) && tree instanceof ConditionalExpressionTree) {
+            AnnotatedTypeMirror atmTrue =
+                    getAnnotatedType(((ConditionalExpressionTree) tree).getTrueExpression());
```

Comment:
I would move this cast into a local variable to avoid the repetition. That might also keep these lines from being too long.